CONTRIBUTING.md

```diff
@@ -1,20 +1,96 @@
+easier to understand a Bug or an Enhancement from the title alone. For example, is the title **"Browsers should support
+blinking text"** a bug or a feature request?
+- Enhancements: The title should be an imperative statement of how things should be. **"Add support for blinking text"**
 - Bugs: The title should be a declarative statement of how things are. **"Text does not blink"**
 
-## Git Commit Guidelines
+## Contributing Code
+
+### Build Dependencies
+
+Before you can build the Cisco Spark JS SDK, you will need the following dependencies:
+- [Node.js](https://nodejs.org/) 6.x (LTS)
+  - We recommend using [nvm](https://github.com/creationix/nvm) (or [nvm-windows](https://github.com/coreybutler/nvm-windows))
+    to easily switch between Node.js versions
+- [Git](https://git-scm.com/)
+
+You will need to create a file called `.env` that defines, at a minimum:
+
+- `CISCOSPARK_CLIENT_ID`
+- `CISCOSPARK_CLIENT_SECRET`
+- `CISCOSPARK_REDIRECT_URI`
+- `CISCOSPARK_SCOPE`
+
+You can get these values by registering a new integration on the [Spark for Developers](https://developer.ciscospark.com/add-integration.html) portal.
+
+### Building the SDK
+
+Fork the [spark-js-sdk](https://github.com/ciscospark/spark-js-sdk/) repository and `git clone` your fork:
+
+    git clone https://github.com/your-username/spark-js-sdk.git
+
+Install tooling dependencies with:
+
+    npm install
+
+Specify which package to build by using the `PACKAGE` environment variable. To build the entire SDK, use `ciscospark`:
+
+    PACKAGE=ciscospark npm run grunt:package -- build
+
+### Running Tests
+
+#### Run All Tests
+    npm run grunt:concurrent -- test
 
-As part of the build process, commits are run through [conventional changelog](https://github.com/conventional-changelog/conventional-changelog) to generate the changelog. Please adhere to the following guidelines when formatting your commit messages.
+#### Run Unit Tests
+Handy during early plugin development when you can write a bunch of unit tests.
 
-### Commit Message Format
+    UNIT_ONLY=true PACKAGE=PACKAGENAME npm run grunt:package -- test
+
+### Run unit tests in watch mode
+OK, this one's a handful and requires a global package, but there were too many possible variants to
+hardcode it any where.
+
+    npm install -g nodemon
+    nodemon -w packages/PACKAGENAME/src -w packages/PACKAGENAME/test -x "UNIT_ONLY=true PACKAGE=PACKAGENAME npm run --silent grunt:package express:test test:node"
+
+#### Run Node.js Tests
+Usually faster, and can build on the fly, thus no need to rebuild everything between test runs
+
+    PACKAGE=PACKAGENAME npm run grunt:package -- express:test test:node
+
+#### Run Browser Tests
+Keeps the browser open so that you can reload set break points and reload the page
+
+    KARMA_DEBUG=true PACKAGE=PACKAGENAME npm run grunt:package -- express:test test:browser
+
+### Git Commit Guidelines
+
```

Comment:
I just discovered https://conventionalcommits.org. We should try to rely on that when possible, but I haven't read through all of it yet.